### PR TITLE
fix(dashboard): chart respects absolute min and max between data and thresholds

### PR DESCRIPTION
**Pull request by hwandersman**
## Overview
When a threshold is defined higher than the max or lower than the min of the yAxis on the line and scatter chart it did not show up.

Updated the following:
* When there are no properties or property data customer can still add/edit/delete thresholds. They will just not be shown on the chart.
* When a threshold is set >max or <min and the y-axis is on "Auto" then the axis will be updated to have at least that max/min with some padding
* Thresholds will not work when multiple y-axis are on the chart. Will need to wait to get feedback about that feature in general and this will not be in scope.
* The default threshold style setting is line + filled region

## Verifying Changes
Verified with unit tests and manual tests.

Some edge cases:
1. yAxis is set manually - manual min and/or max is always prioritized and observed
2. yAxis is auto, there is data and no thresholds, yAxis is automatically set by echarts
3. yAxis is auto, there is no data and thresholds, nothing show up on the chart
4. yAxis is auto, there is data and thresholds, absolute min > 0, absolute max > data, yAxis max is updated and padded to the nearest whole num / multiple power of 10
5. yAxis is auto, there is data and thresholds, absolute min < data, absolute max < 0, yAxis min is updated and padded to the nearest whole num / multiple power of 10
6. yAxis is auto, there is data and thresholds, absolute min < 0, absolute max > data, yAxis min and max are updated and padded to the nearest whole num / multiple power of 10
7. Absolute min / max is a decimal, padded correctly
8. Absolute min / max is a negative num, padded correctly
9. Absolute min / max is a very large num, padded correctly

https://github.com/awslabs/iot-app-kit/assets/87385528/a658a005-6150-4cae-a112-555ce9fd3276

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/react-components/src/components/chart/converters/padYAxis.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import { DataPoint, DataStream, Threshold } from '@iot-app-kit/core';
```
Comment (corteggiano) [inline packages/react-components/src/components/chart/converters/padYAxis.ts:1]: note for other reviewers, this is pulled directly from the synchro-charts codebase :) reduce reuse recycle

**Review by corteggiano — APPROVED** [review]
